coalib/bearlib/abstractions/Linter.py

```diff
@@ -122,6 +122,15 @@ def __repr__(cls):
                 cls.__name__, options["executable"])
 
     class LinterBase(LocalBear, metaclass=LinterMeta):
+        def __init__(self):
+            if options["output_format"] == "regex":
+                options["output_regex"] = re.compile(options["output_regex"])
+                if options["output_regex"].groups > 8:
+                    self.warn("Performance: Using unnecessary capture groups"
+                              " affects the performance of coala. "
+                              "You should use ``(?:<pattern>)`` instead of"
+                              " ``(<pattern>)`` for your regex")
+
```

Comment:
@Makman2 Is this even close to what we discussed a few days ago? I wrote this today, thats why difference may occur. :worried: